cmd/argocd/commands/app.go

```diff
@@ -1026,7 +1026,7 @@ func findandPrintDiff(ctx context.Context, app *argoappv1.Application, proj *arg
 	items := make([]objKeyLiveTarget, 0)
 	if diffOptions.local != "" {
 		localObjs := groupObjsByKey(getLocalObjects(ctx, app, proj, diffOptions.local, diffOptions.localRepoRoot, argoSettings.AppLabelKey, diffOptions.cluster.Info.ServerVersion, diffOptions.cluster.Info.APIVersions, argoSettings.KustomizeOptions, argoSettings.ConfigManagementPlugins, argoSettings.TrackingMethod), liveObjs, app.Spec.Destination.Namespace)
-		items = groupObjsForDiff(resources, localObjs, items, argoSettings, app.InstanceName(argoSettings.ControllerNamespace))
+		items = groupObjsForDiff(resources, localObjs, items, argoSettings, app.InstanceName(argoSettings.ControllerNamespace), app.Spec.Destination.Namespace)
```

Comment:
The use of `app.InstanceName(argoSettings.ControllerNamespace)` as `appName` is inconsistent with the other calls to `groupObjsForDiff()` and odd to me. I cannot wrap my head around to tell if it is legacy or a mistake

https://github.com/argoproj/argo-cd/blob/4ecffdd236bffe524538b39b293de19f905e762d/pkg/apis/application/v1alpha1/types.go#L2971-L2982

If I change it to `app.Name` integration tests fail, not sure if it'll be better fixing them or leave it as:

https://github.com/argoproj/argo-cd/actions/runs/5191714335/jobs/9359975651

Shouldn't they all build the same app name one way or another?

Reply:
Fixed by #13980